lib/ansible/modules/system/mount.py

```diff
@@ -551,23 +539,21 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            boot=dict(default='yes', choices=['yes', 'no']),
-            dump=dict(),
-            fstab=dict(default=None),
-            fstype=dict(),
-            path=dict(required=True, aliases=['name'], type='path'),
-            opts=dict(),
+            boot=dict(type='bool', default=True),
+            dump=dict(type='str'),
```

Comment:
This is not required as `type=str` is the default type. The same bellow.

Reply:
Well, for the type I think it's better to keep it, to make it obvious. In the past the default type was "raw".

Reply:
There was multiple changes in the past but you don't make any backward compatibility fixes for that neither so I don't see a reason why to make exceptions for `type='str'` here.

Reply:
Strict typing in this case seems favorable.